src/Workspaces/Core/Portable/Execution/ChecksumTreeCollection.TreeNodes.cs

```diff
@@ -0,0 +1,259 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Execution
+{
+    /// <summary>
+    /// This is collection of root checksum tree node
+    /// </summary>
+    internal partial class ChecksumTreeCollection
+    {
+        /// <summary>
+        /// root tree node of checksum tree
+        /// </summary>
+        private sealed class RootTreeNode : SubTreeNode, IRootChecksumTreeNode
```

Comment:
added RootTreeNode. except that same code as before. just separated to its own file.